sqlstore/src/main/resources/db/migration_beforeMigrate/05_standard.sql

```diff
@@ -430,21 +430,25 @@ CREATE PROCEDURE addIdentity(
   iSampleType varchar(255)
 ) BEGIN
   DECLARE newSampleId bigint(20);
+  DECLARE errorMessage varchar(300);
   
-  IF NOT EXISTS (SELECT 1 FROM Identity WHERE externalName = iExternalName) THEN
-    INSERT INTO Sample(name, alias, description, sampleType, project_projectId, scientificName, lastModifier, securityProfile_profileId) VALUES 
-        ('TEMP_IDENT', iAlias, iDescription, iSampleType, (SELECT projectId FROM Project WHERE shortName = iProject), iScientificName, getAdminUserId(), (SELECT securityProfile_profileId FROM Project WHERE shortName = iProject));
-    SET newSampleId = LAST_INSERT_ID();
+  IF EXISTS (SELECT 1 FROM Identity JOIN Sample USING (sampleId) JOIN Project ON Project.projectId = Sample.project_projectId WHERE externalName = iExternalName AND Project.alias = iAlias) THEN
```

Comment:
This checks the case where an identical identity has already been entered, but doesn't fix the sub-cases where a) the lab tells us to enter `DYS_8001` on two separate occasions, or b) the lab tells us to enter an identity with external name `EXT1` for both `DYS_8001` and `DYS_5001`.